featurebyte/models/feature.py

```diff
@@ -7,213 +7,46 @@
-    node_name: str
-        Node name of the graph which represent the feature
-    tabular_source: TabularSource
-        Tabular source used to construct this feature
-    readiness: FeatureReadiness
-        Feature readiness
-    version: VersionIdentifier
-        Feature version
-    online_enabled: bool
-        Whether to make this feature version online enabled
-    definition: str
-        Feature definition
-    entity_ids: List[PydanticObjectId]
-        Entity IDs used by the feature
-    table_ids: List[PydanticObjectId]
-        Table IDs used by the feature
-    primary_table_ids: Optional[List[PydanticObjectId]]
-        Primary table IDs of the feature (auto-derive from graph)
-    feature_namespace_id: PydanticObjectId
-        Feature namespace id of the object
-    feature_list_ids: List[PydanticObjectId]
-        FeatureList versions which use this feature version
-    deployed_feature_list_ids: List[PydanticObjectId]
-        Deployed FeatureList versions which use this feature version
-    created_at: Optional[datetime]
-        Datetime when the Feature was first saved
-    updated_at: Optional[datetime]
-        When the Feature get updated
+    BaseFeatureTargetModel is the base class for FeatureModel & TargetModel.
+    It contains all the attributes that are shared between FeatureModel & TargetModel.
     """
 
     dtype: DBVarType = Field(allow_mutation=False, default=DBVarType.UNKNOWN)
     node_name: str
     tabular_source: TabularSource = Field(allow_mutation=False)
-    readiness: FeatureReadiness = Field(allow_mutation=False, default=FeatureReadiness.DRAFT)
     version: VersionIdentifier = Field(allow_mutation=False, default=None)
-    online_enabled: bool = Field(allow_mutation=False, default=False)
     definition: Optional[str] = Field(allow_mutation=False, default=None)
 
     # special handling for those attributes that are expensive to deserialize
     # internal_* is used to store the raw data from persistence, _* is used as a cache
     internal_graph: Any = Field(allow_mutation=False, alias="graph")
     _graph: Optional[QueryGraph] = PrivateAttr(default=None)
 
-    # list of IDs attached to this feature
+    # list of IDs attached to this feature or target
     entity_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     primary_table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
-    feature_namespace_id: PydanticObjectId = Field(allow_mutation=False, default_factory=ObjectId)
-    feature_list_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
-    deployed_feature_list_ids: List[PydanticObjectId] = Field(
-        allow_mutation=False, default_factory=list
-    )
     user_defined_function_ids: List[PydanticObjectId] = Field(
         allow_mutation=False, default_factory=list
     )
```

Comment:
@jevonyeoh @YS-L 
These are the common attributes that should share between target & feature. Does it make sense to you?

Reply:
makes sense to me!